rest/src/main/scala/io/udash/rest/openapi/OpenApi.scala

```diff
@@ -0,0 +1,599 @@
+package io.udash
+package rest.openapi
+
+import com.avsystem.commons._
+import com.avsystem.commons.misc.{AbstractValueEnum, AbstractValueEnumCompanion, EnumCtx}
+import io.udash.rest.JsonValue
+import com.avsystem.commons.serialization.json.JsonStringOutput
+import com.avsystem.commons.serialization.{transientDefault => td, _}
+
+/**
+  * Representation of
+  * [[https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.1.md#openapi-object OpenAPI Object]]
+  * from [[https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.1.md OpenAPI 3.0 specification]].
+  * It may be serialized to OpenAPI 3.0 compliant JSON using
+  * [[com.avsystem.commons.serialization.json.JsonStringOutput JsonStringOutput]].
+  * This JSON can then be consumed by tools that support OpenAPI 3.0, e.g.
+  * [[https://swagger.io/tools/swagger-ui/ Swagger UI]].
+  */
+case class OpenApi(
+  openapi: String = OpenApi.Version,
+  info: Info,
+  paths: Paths,
+  @td servers: List[Server] = Nil,
+  @td components: OptArg[Components] = OptArg.Empty,
+  @td security: List[SecurityRequirement] = Nil,
+  @td tags: List[Tag] = Nil,
+  @td externalDocs: OptArg[ExternalDocumentation] = OptArg.Empty
+)
+object OpenApi extends HasGenCodec[OpenApi] {
+  final val Version = "3.0.1"
```

Comment:
It's `3.0.2` now.

Reply:
The code I wrote based on 3.0.1, so we can assume this is the version generated ;)